msal-dotnet-articles/advanced/managed-identity.md

```diff
@@ -1,51 +1,79 @@
+AuthenticationResult result = await mi.AcquireTokenForManagedIdentity(resource)
     .ExecuteAsync()
     .ConfigureAwait(false);
 ```
 
-* The scopes array should contain a single scope as managed identity acquires token for a resource.
-* The parameter userAssignedClientIdOrResourceId can either contain the client id of the user assigned managed identity or the resource id in case the client id is not yet available.
-* For system assigned managed identity the parameter userAssignedClientIdOrResourceId need not be passed.
-* For failed requests, the error response contains a correlation id that can be used for further investigation. The MSAL's correlation id generated in MSAL or passed in to MSAL is different than the one returned in server error response as MSAL cannot pass the correlation id to managed identity token acquisition endpoints.
+### User-assigned managed identities
+
+For user-assigned managed identities, the developer needs to pass the managed identity client ID or the full resource identifier string if client ID is not available when creating <xref:Microsoft.Identity.Client.IManagedIdentityApplication>.
+
+Like in the case for system-assigned managed identities, <xref:Microsoft.Identity.Client.IManagedIdentityApplication.AcquireTokenForManagedIdentity(System.String)> is called with the resource to acquire a token for, such as `https://management.azure.com`.
+
+```csharp
+IManagedIdentityApplication mi = ManagedIdentityApplicationBuilder.Create(userAssignedId)
+    .WithExperimentalFeatures()
+    .Build();
+
+AuthenticationResult result = await mi.AcquireTokenForManagedIdentity(resource)
+    .ExecuteAsync()
+    .ConfigureAwait(false);
+```
 
-## Supported by MSAL
+## Caching
 
-MSAL supports the following sources for managed identity
+By default, MSAL.NET supports in-memory caching. To explore additional caching options or implement a custom cache, see [Token cache serialization in MSAL.NET](/azure/active-directory/develop/msal-net-token-cache-serialization). We do not recommend sharing a cache between two or more Azure sources with managed identities enabled as this can result in the same token being shared across the resources.
 
-Supported:
+## Troubleshooting
 
-* App services
-* IMDS (VMs)
-* Azure Arc
+For failed requests the error response contains a correlation ID that can be used for further diagnostics and log analysis. Keep in mind that the correlation IDs generated in MSAL.NET or passed into MSAL are different than the one returned in server error responses, as MSAL.NET cannot pass the correlation ID to managed identity token acquisition endpoints.
 
-In Progress:
+## Potential errors
 
-* Cloud Shell
-* Service Fabric
+### `MsalServiceException` Error Code: `managed_identity_failed_response` Error Message: An unexpected error occurred while fetching the AAD token
```

Comment:
nit: I would make Potential errors a sub section of troubleshooting, and error sections smaller heading for easier reading.